File: Practicing/Java NetBeans/ArbolBinario/src/main/java/arbolbinario/Arbol.java

package arbolbinario;

public class Arbol {
    
    Nodo arbol = null;
    int elementos;
    
    public Arbol(){
        this.arbol = null;
        this.elementos = 0;
    }
    
    public Arbol(int value){
        this.arbol = new Nodo(value);
        this.elementos = 1;
    }
    
    public int getSize(){
        return this.elementos;
    }
    
    public boolean isEmpty(){
        return this.arbol == null;
    }
    
    public void insert(int value){
        if(this.isEmpty() == true){
            
            this.arbol = new Nodo(value);
            this.arbol.setPadre(null);
            this.elementos ++;
            
            System.out.println("Nodo agregado correctamente!!");
            
        } else {
            this.auxInsert(this.arbol, value, this.arbol);
        }
    }
    
    private void auxInsert( Nodo arbol, int value, Nodo padre ){
        
        if(value > arbol.getData()){
            
            if(arbol.getDerecha() == null){
                
                arbol.setDerecha(new Nodo(value));
                arbol.getDerecha().setPadre(padre);
                
                System.out.println("Nodo agregado correctamente!!");
                
                this.elementos ++;
                
            } else {
                
                auxInsert(arbol.getDerecha(), value, arbol);
                
            }
            
        } else {
            
            if(arbol.getIzquierda() == null){
                
                arbol.setIzquierda(new Nodo(value));
                arbol.getIzquierda().setPadre(padre);
                
                System.out.println("Nodo agregado correctamente!!");
                
                this.elementos ++;
                
            } else {
                
                auxInsert(arbol.getIzquierda(), value, arbol);
                
            }
            
        }
        
    }
    
    public void preOrden(){
        this.auxPreOrden(this.arbol);
        System.out.println("");
    }
    
    private void auxPreOrden( Nodo arbol ){
        if(arbol == null){
            return;
        } else {
            System.out.print("[" + arbol.getData() + "]");
            auxPreOrden(arbol.getIzquierda());
            auxPreOrden(arbol.getDerecha());
        }
    }
    
    public void inOrden(){
        this.auxInOrden(this.arbol);
        System.out.println("");
    }
    
    private void auxInOrden( Nodo arbol ){
        if(arbol == null){
            return;
        } else {
            auxInOrden(arbol.getIzquierda());
            System.out.print("[" + arbol.getData() + "]");
            auxInOrden(arbol.getDerecha());
        }
    }
    
    public void postOrden(){
        this.auxPostOrden(this.arbol);
        System.out.println("");
    }
    
    private void auxPostOrden( Nodo arbol ){
        if(arbol == null){
            return;
        } else {
            auxPostOrden(arbol.getIzquierda());
            auxPostOrden(arbol.getDerecha());
            System.out.print("[" + arbol.getData() + "]");
            
        }
    }
    
    public void showThree(){
        this.auxShowThree(this.arbol, 0);
    }
    
    private void auxShowThree( Nodo arbol, int contador ){
        if(arbol == null){
		return;
	} else {
		auxShowThree(arbol.getDerecha(), contador+1);
		for(int i = 0; i < contador; i++){
                    System.out.print("   ");
		}
		System.out.println(arbol.getData());
		auxShowThree(arbol.getIzquierda(), contador+1);
	}
    }
    
}

File: Practicing/Python/ArbolBinario/Arbol.py
from Nodo import Nodo

class Arbol:
    def __init__(self, value = None):
        if value != None:
            self.__arbol = Nodo(value)
            self.__size = 1
        else:
            self.__arbol = None
            self.__size = 0

    def getSize(self):
        return self.__size

    def isEmpty(self):
        return self.__arbol == None

    def insert(self, value):
        if self.__arbol == None:
            self.__arbol = Nodo(value)
            self.__size += 1
        else:
            self.__auxInsert(self.__arbol, value)

    def __auxInsert(self, arbol, value):
        if value > arbol.getData():
            if arbol.getDerecha() == None:
                arbol.setDerecha(Nodo(value))
                self.__size += 1
            else:
                self.__auxInsert(arbol.getDerecha(), value)
        else:
            if arbol.getIzquierda() == None:
                arbol.setIzquierda(Nodo(value))
                self.__size += 1
            else:
                self.__auxInsert(arbol.getIzquierda(), value)

    def preOrden(self):
        self.__auxPreOrden(self.__arbol)
        print()

    def __auxPreOrden(self, arbol):
        if arbol == None:
            return
        else:
            print("[", arbol.getData(), "]", end="")
            self.__auxPreOrden(arbol.getIzquierda())
            self.__auxPreOrden(arbol.getDerecha())

    def inOrden(self):
        self.__auxInOrden(self.__arbol)
        print()

    def __auxInOrden(self, arbol):
        if arbol == None:
            return
        else:
            self.__auxInOrden(arbol.getIzquierda())
            print("[", arbol.getData(), "]", end="")
            self.__auxInOrden(arbol.getDerecha())

    def postOrden(self):
        self.__auxPostOrden(self.__arbol)
        print()

    def __auxPostOrden(self, arbol):
        if arbol == None:
            return
        else:
            self.__auxPostOrden(arbol.getIzquierda())
            self.__auxPostOrden(arbol.getDerecha())
            print("[", arbol.getData(), "]", end="")